From 0800741a78086db86aec75a2ec42197cd7a81a94 Mon Sep 17 00:00:00 2001
From: Ferdinand Xu <cheng.a.xu@intel.com>
Date: Sun, 26 Jul 2015 22:32:59 -0400
Subject: [PATCH 0691/1164] CDH-42369 : HIVE-11352: Avoid the double
 connections with 'e' option[beeline-cli
 branch](Ferdinand Xu via Xuefu Zhang)

Change-Id: I8777cdbe384b52d41f3d0336eae2168257b725bb
---
 .../src/java/org/apache/hive/beeline/BeeLine.java  |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLine.java b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
index 8034c7e..1fce0ab 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLine.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
@@ -872,10 +872,10 @@ public int begin(String[] args, InputStream inputStream) throws IOException {
         }
       } else {
         int code = initArgsFromCliVars(args);
-        defaultConnect(false);
-        if (code != 0){
+        if (code != 0 || exit) {
           return code;
         }
+        defaultConnect(false);
         getOpts().updateBeeLineOptsFromConf();
         getOpts().setShowHeader(false);
         getOpts().setOutputFormat("dsv");
-- 
1.7.9.5

